tensorflow/python/estimator/inputs/queues/feeding_functions.py

```diff
@@ -218,6 +220,50 @@ def __call__(self):
     return feed_dict
 
 
+class _GeneratorFeedFn(object):
+    """Creates feed dictionaries from `Generator` of `dicts` of numpy arrays."""
+
+    def __init__(self,
+                 placeholders,
+                 generator,
+                 batch_size,
+                 random_start=False,
+                 seed=None,
+                 num_epochs=None):
+        first_sample = next(generator())
+        if len(placeholders) != len(first_sample):
+            raise ValueError("Expected {} placeholders; got {}.".format(
+                len(first_sample), len(placeholders)))
+        self._col_placeholders = placeholders
+        self._generator_function = generator
+        self._iterator = generator()
+        self._batch_size = batch_size
+        self._num_epochs = num_epochs
+        self._epoch = 0
+        random.seed(seed)
+
+    def __call__(self):
+        if self._num_epochs and self._epoch >= self._num_epochs:
+            raise errors.OutOfRangeError(
+                None, None, "Already emitted %s epochs." % self._epoch)
+        list_dict = {}
+        list_dict_size = 0
+        while list_dict_size < self._batch_size:
+            try:
+                data_row = next(self._iterator)
+            except StopIteration:
+                self._epoch += 1
+                self._iterator = self._generator_function()
+                data_row = next(self._iterator)
+            for index, key in enumerate(sorted(data_row.keys())):
+                list_dict.setdefault(self._col_placeholders[index],
+                                     list()).append(data_row[key])
+            list_dict_size += 1
+        feed_dict = {key: np.asarray(item)
+                        for key, item in list(list_dict.items())}
```

Comment:
I would love some check that emits a very obvious error if the dicts emitted by the generator do not have identical key sets.